### PR TITLE
Vendors: Retry getting Amazon data from the affiliate server

### LINKED ISSUE #5474: Fix /isbn to retry Affiliate API up to 3 times (w/ 1 sec timeout)
Should be fixed in #6332

Right now if one goes to a page like https://openlibrary.org/isbn/9781797210285 and the book isn't in OL, we enqueue a request for it in the Affiliate Server.

However, after enqueued, Open Library will return a Not Found page (and import the book behind the scenes).

What we should do is if one hits the /isbn endpoint directly, that we should assume they're willing for the import to complete and we should retry up to e.g. 3 times.

### Relevant url?

https://openlibrary.org/isbn/9781797210285

### Proposal & Constraints

Alternatively, we could have GET parameter which lets people skip the retry (if they're trying to just submit books for import)

### Related files

https://github.com/internetarchive/openlibrary/blob/master/openlibrary/plugins/openlibrary/code.py#L374

https://github.com/internetarchive/openlibrary/blob/master/openlibrary/core/vendors.py#L340

### Stakeholders

@jimchamp @cclauss

### PR DESCRIPTION
**Pull request by cclauss**
Closes #5474 

The /isbn endpoint should retry a few times to see if the __affiliate service__ has the Amazon result in the cache.  If this import is already queued up, the set operation will ensure that it will be ignored automatically.

### CODE REVIEW

**Comment by cdrini** [comment]
Can you search the history to see if we can do `list | str` ?  I vaguely recall we couldn't for some reason... There's a PR somewhere I think. Might need `Union[list, str]`

**Comment by cclauss** [comment]
Did you mean to **close** this issue?

https://github.com/internetarchive/openlibrary/pull/6320 says that the type hints thing is about **_Cython use in Solr_**.
* Confirmed by https://cython.readthedocs.io/en/latest/src/tutorial/pure.html#typing-module

Also see:
* https://github.com/internetarchive/openlibrary/pull/5922
* https://github.com/internetarchive/openlibrary/blob/master/scripts/solr_builder/solr_builder/fn_to_cli.py

**Comment by cdrini** [comment]
Ooops!!! No I did not meant to close it :P

File: openlibrary/core/vendors.py [hunk 1]
```diff
@@ -295,13 +299,22 @@ def search_amazon(title='', author=''):
...
+    retries: int = 3,
+    sleep_sec: float = 0.1,
+) -> dict | None:
     """Uses the Amazon Product Advertising API ItemLookup operation to locatate a
     specific book by identifier; either 'isbn' or 'asin'.
     https://docs.aws.amazon.com/AWSECommerceService/latest/DG/ItemLookup.html
 
     :param str id_: The item id: isbn (10/13), or Amazon ASIN.
     :param str id_type: 'isbn' or 'asin'.
+    :param ??? resources: ???
```
Comment (cclauss) [inline openlibrary/core/vendors.py:315]:
```suggestion
    :param resources: Used for AWSE Commerce Service lookup -- See Amazon docs
```

**Comment by cclauss** [comment]
Test this on one of the VMs like `testing`.
* Type hints
* Comments

**Comment by cclauss** [comment]
# Ready for review.

**Pull request merged**